receiver/prometheusreceiver/metrics_receiver_helper_test.go

```diff
@@ -140,22 +165,38 @@ func setupMockPrometheus(tds ...*testData) (*mockPrometheus, *promcfg.Config, er
 	configP["scrape_configs"] = jobs
 	cfg, err := yaml.Marshal(&configP)
 	if err != nil {
-		return mp, nil, err
-	}
-	// update attributes value (will use for validation)
-	for _, t := range tds {
-		t.attributes = pdata.NewAttributeMap()
-		t.attributes.Insert("service.name", pdata.NewAttributeValueString(t.name))
-		t.attributes.Insert("host.name", pdata.NewAttributeValueString(host))
-		t.attributes.Insert("job", pdata.NewAttributeValueString(t.name))
-		t.attributes.Insert("instance", pdata.NewAttributeValueString(u.Host))
-		t.attributes.Insert("port", pdata.NewAttributeValueString(port))
-		t.attributes.Insert("scheme", pdata.NewAttributeValueString("http"))
+		return nil, err
 	}
 	pCfg, err := promcfg.Load(string(cfg), false, gokitlog.NewNopLogger())
-	return mp, pCfg, err
-}
 
+	//check for customConfig requirement and edit default receiver config accordingly
+	if promConfig != nil {
+		if promConfig.externalLabels != nil {
+			pCfg.GlobalConfig.ExternalLabels = promConfig.externalLabels
+		}
+		if !promConfig.honorTimestamp {
+			for _, scrapeConfig := range pCfg.ScrapeConfigs {
+				scrapeConfig.HonorTimestamps = false
+			}
+		}
+		if promConfig.honorLabel {
+			for _, scrapeConfig := range pCfg.ScrapeConfigs {
+				scrapeConfig.HonorLabels = true
+			}
+		}
+		if promConfig.labelLimit > 0 {
+			for _, scrapeConfig := range pCfg.ScrapeConfigs {
+				scrapeConfig.LabelLimit = promConfig.labelLimit
+			}
+		}
+		if promConfig.renamingCfg != nil {
+			for _, scrapeConfig := range pCfg.ScrapeConfigs {
+				scrapeConfig.MetricRelabelConfigs = promConfig.renamingCfg
+			}
+		}
+	}
```

Comment:
This is going to be cumbersome and error-prone to maintain.  I'd prefer to use a mutator that can be passed to `testComponent()`.  See how I did it in [this PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6421/files#diff-ad6f97323c1aea65ad2748924e71b6f180a249d67a68151534874f1b0965d1e9R502), which might remove the need for these changes anyway when it lands.

Reply:
@Aneurysm9 thanks. I have now updated accordingly.